src/inmanta/data/__init__.py

```diff
@@ -5303,17 +5340,23 @@ async def insert(self, connection: Optional[asyncpg.connection.Connection] = Non
         self.make_hash()
         await super().insert(connection=connection)
         # TODO: On conflict or is not exists or just make every update an upsert?
+        resource_status = (
+            state.ResourceStatus.UNDEFINED if self.status is const.ResourceState.undefined else state.ResourceStatus.HAS_UPDATE
```

Comment:
> else state.ResourceStatus.HAS_UPDATE

We set `resource_status` to `state.ResourceStatus.HAS_UPDATE` for all the following resource states ?
``` 
    unavailable = "unavailable"  # This state is set by the agent when no handler is available for the resource
    skipped = "skipped"  #
    dry = "dry"
    deployed = "deployed"
    failed = "failed"
    deploying = "deploying"
    available = "available"
    cancelled = "cancelled"  # When a new version is pushed, in progress deploys are cancelled
    skipped_for_undefined = "skipped_for_undefined" 
```

Reply:
This is the code path where a new resources are pushed to the server (put_version or put_partial) and only those that are not known by the server yet. So only the "initial" states are relevant here. All states that are the result of a deployment action cannot happen. So the relevant ones are: `available`, `undefined` and `skipped_for_undefined`. So it it's not an UNDEFINED resource, it must be HAS_UPDATE.